Jenkinsfile

```diff
@@ -0,0 +1,153 @@
+                        checkout scm
+                        echo 'Within build on Ubuntu16.04'   
+                            sh 'sudo apt-get update && sudo apt-get install -y devscripts debhelper python-all-dev python-pbr python-setuptools'
+                            sh 'debuild --no-tgz-check clean binary'
+                        dir("${WORKSPACE}/debs/cloud-info-provider-openstack") {
+                            sh 'debuild --no-tgz-check clean binary'
+                        }
+                        dir("${WORKSPACE}/debs/cloud-info-provider-opennebula") {
+                            sh 'debuild --no-tgz-check clean binary'
+                        }
+                    }
+                    post {
+                        success {
+                            archiveArtifacts artifacts: '**/debs/*.deb'                        }
+                    }
+                }
+                stage('Build on CentOS7') {
+                    agent {
+                        label 'bcentos7'
+                    }
+                    steps {
+                        checkout scm
+                        echo 'Within build on CentOS7'
+                        sh 'sudo yum install -y rpm-build centos-release-openstack-newton python-pbr python-setuptools'
+                        sh 'python setup.py sdist'
+                        sh 'mkdir ~/rpmbuild'
+                        sh "echo '%_topdir %(echo $HOME)/rpmbuild' > ~/.rpmmacros"
+                        sh 'mkdir -p ~/rpmbuild/{SOURCES,SPECS}'
+                        sh 'cp dist/cloud_info_provider*.tar.gz ~/rpmbuild/SOURCES/'
+                        sh 'cp rpm/cloud-info-provider*.spec ~/rpmbuild/SPECS/'
+                        //sh 'rpmbuild -ba ~/rpmbuild/SPECS/cloud-info-provider.spec'
+                        sh 'rpmbuild -ba ~/rpmbuild/SPECS/cloud-info-provider-openstack.spec'
+                        sh 'rpmbuild -ba ~/rpmbuild/SPECS/cloud-info-provider-opennebula.spec'
+                        sh 'cp ~/rpmbuild/SRPMS/*.rpm ~/rpmbuild/RPMS/noarch/*.rpm ${WORKSPACE}/rpm/'
+                    }
+                    post {
+                        success {
+                            archiveArtifacts artifacts: '**/rpm/*.rpm'
+                        }
+                    }
+                }
+            }
+        }
+        
+        stage('Docker container building (git tagged version)') {
+            when {
+                anyOf {
+                    branch 'master'
+                    buildingTag()
+                }
+            }
+            steps {
+                script {
+                    if (env.BRANCH_NAME == 'master') {
+                        IMAGE_ID = env.docker_repo + ":latest"
+                    }
+                    else {
+                        IMAGE_ID = env.docker_repo + env.TAG_NAME
+                    }
+                }
+                echo "Building Docker image ${IMAGE_ID}"
+            }
+            post {
```

Comment:
I would put here a success{ } post phase to ship artefacts to the appropriate repo so that you have continuous delivery.